Yoast/wordpress-seo
# Replace deprecated filter `block_categories` by `block_categories_all`

## Context


* From WordPress 5.8.0 onwards, the `block_categories` filter [will be deprecated](https://github.com/WordPress/WordPress/blob/master/wp-includes/block-editor.php#L85).

## Summary


This PR can be summarized in the following changelog entry:

* Introduces a WordPress helper to get the WordPress version number.
* Introduces the `block_categories_all` filter as an alternative to `block_categories` for users with WordPress 5.8 or higher.

## Relevant technical choices:

* Had to implement the code using a conditional to ensure that the plugin keeps working for users who don't use WordPress 5.8 (yet).
* Just changing the filter name seems to be enough. While the [filter's second argument] (https://github.com/WordPress/WordPress/blob/master/wp-includes/block-editor.php#L85) also changes, this doesn't impact our code since [we're not using this argument](https://github.com/Yoast/wordpress-seo/blob/trunk/src/integrations/blocks/block-categories.php#L32).
* Unrelated to the issue, I added all missing helpers to the helpers surface.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Use the WordPress Beta Tester plugin to set your site to WordPress 5.8.
    * The actual updating should be triggered under Dashboard > Updates.

_Replicate the problem_
* On `trunk` / the 16.6 zip, see the following deprecation warning when creating a new post. You may see it flash by briefly in the top of the screen. Alternatively, you can view the page source to see the warning.
    * `block_categories is deprecated since version 5.8.0! Use block_categories_all instead`.
   
_Test the solution_

* Now check-out this branch, or upload the 16.7 zip.
* Confirm that the deprecation warning is gone.  
* Confirm that the block categories still work as expected.
    * In a post, click the top left block inserter. See the `Yoast Structured Data Blocks` and `Yoast Internal Linking Blocks` block categories. 
* Downgrade your site to WordPress 5.7 using the WordPress Beta Tester plugin.
* Confirm that the block categories still work as expected.

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes #
